src/compas/utilities/itertools_.py

```diff
@@ -126,18 +126,24 @@ def repeatfunc(func, times=None, *args):
     return starmap(func, repeat(args, times))
 
 
-def pairwise(iterable):
+def pairwise(iterable, wrap=False):
     """s -> (s0,s1), (s1,s2), (s2, s3), ..."""
+    iterable_type = type(iterable)
+    if wrap:
+        iterable = iterable + iterable_type(iterable[:1])
```

Comment:
I think this doesn't work for many cases, e.g. a generator cannot be sliced this way, and it cannot be re-constructed either.

Reply:
also agreed

Reply:
true, I tried to find a not too ugly way to make it work

Reply:
i think this can just be the responsibility of the user

```python
for u, v in pairwise(vertices + vertices[0:1]):
    print(u, v)
```

Reply:
ok I removed the wrap option in pairwise and function and I updated the mesh property functions